### ISSUE TITLE
Basic Login via js api not working

### ISSUE THREAD

**Issue by notanark** [ParsePlatform/Parse-SDK-JS#262]
I don't thing it's me?

`var fs = require('fs'),
https = require('https');
var ssl = {
key: fs.readFileSync('./', 'utf8'),
cert: fs.readFileSync('./', 'utf8'),
ca: [fs.readFileSync('./', 'utf8'),
fs.readFileSync('./', 'utf8')]
};
var express = require('express');
var ParseServer = require('parse-server').ParseServer;

var databaseUri = ' ';

var port = process.env.PORT || 4337;
//var port = 443;
var api = new ParseServer({
databaseURI: databaseUri,
cloud: __dirname + '/cloud/main.js',
serverURL: 'aa',

appId: 'aa,
'javascriptKey':'aa',
restAPIKey: 'a',
clientKey:'a',
facebookAppIds :['a'], 
masterKey: 'a'
});
var app = express();

// Serve the Parse API on the /parse URL prefix
var mountPath = '/a';
app.use(mountPath, api);
app.use(express.static('public'));
// Parse Server plays nicely with the rest of your web routes
app.get('/', function(req, res) {
res.status(200).send('I dream of being a web site.');
});

app.listen(4338,function () { console.log('Server on port 4338'); });
https.createServer(ssl, app)
.listen(port, function() {
console.log('Server running on port ' + port + '.');
});`

Response is {"error":"unauthorized"}

My appid is correct.

**Comment by notanark** [comment]
Cross issue: https://github.com/ParsePlatform/parse-server/issues/1615

**Comment by notanark** [comment]
bump

**Comment by drew-gross** [comment]
We have lots of passing test cases that cover these features. Can you post some code that reliably reproduces your issue? Ideally as a test case we can add to our test suite.

**Comment by notanark** [comment]
I'm literally using the out of box example on a server that already works with IOS. I can pm you the site if you want to have a gander.

**Comment by notanark** [comment]
I've tried multiple things even using the master key. The only thing I can think that it could possibly be even though it sounds stupid is that I'm self hosting the js so it won't get cross domain problems but I am using parse-latest from the cdn. Downloaded yesterday.

**Comment by andrewimm** [comment]
The SDK won't encounter cross-domain issues, since it uses special POST requests with embedded methods.

Can you provide the code you're running with the Parse SDK (not Parse Server) that produces the `{"error": "unauthorized"}` you mention?

**Comment by notanark** [comment]
`
var user = new Parse.User();
user.set("username", "my name");
user.set("password", "my pass");
user.set("email", "email@example.com");

// other fields can be set just like with Parse.Object
user.set("phone", "415-392-0202");

user.signUp(null, {
  success: function(user) {
    console.log(user);
  },
  error: function(user, error) {
    // Show the error message somewhere and let the user try again.
    console.log(error);
  }
});


`

**Comment by andrewimm** [comment]
Are you pointing your Parse SDK at your Parse Server location?

```js
Parse.initialize("YOUR_APP_ID");
Parse.serverURL = 'http://your.server.url:port/path'
```

**Comment by notanark** [comment]
absolutely.

**Comment by notanark** [comment]
I've even tried it with master key, js key, etc. ...

**Comment by andrewimm** [comment]
So, the response you're receiving
```
{"error":"unauthorized"}
```
results when you send the wrong keys to the server. I'd double-check that
a) you're actually communicating with your server, and not `parse.com`
b) you're using the right keys

If our out-of-the box experience were broken, we would have tests that caught that. The error is somewhere in your configuration. Sorry.

**Comment by notanark** [comment]
I can see my server in the chrome net console. It's correct.

**Comment by notanark** [comment]
What it's pointing to is the latest version of parse-server is broken. I have an older different version running on a separate machine with the same exact config that doesn't have these errors.

**Comment by andrewimm** [comment]
Your iOS app points at the new server and works, right?

**Comment by notanark** [comment]
My ios points at a clone of the same server with the same config. The difference is I ran parse update on this new box.

**Comment by andrewimm** [comment]
Unfortunately, unless you can provide a more specific reproducible case, there's nothing I can do to help.

This does not sound like a bug within the JS SDK (you say yourself that changing the parse-server breaks things, not the SDK), so I am closing this issue.

**Comment by hydex86** [comment]
Hi, I know this is an old bug, but I'm experiencing the same problem. The point is everything works well if my parse-server is version 2.2.23 or below. But if I upgrade my server to version 2.2.24 my JS client gets this response: {"error": "unauthorized"}

My iOS client works fine both in version 2.2.23 and 2.2.24. I've tested with parse-server versions 2.2.24 and 2.3.1.

I had to rollback my parse-server version to 2.2.23 to keep working.
